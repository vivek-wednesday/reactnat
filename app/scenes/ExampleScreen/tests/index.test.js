/**
 *
 * Tests for ExampleScreen
 *
 */

import React from 'react';
import { renderProvider } from 'app/utils/testUtils';
import { ExampleScreenTest } from '../index';

describe('<ExampleScreen /> Container tests', () => {
  let submitSpy;

  beforeAll(() => {
    submitSpy = jest.fn();
    jest.mock('./__mocks__/react-native-splash-screen');
  });

  it('should render and match the snapshot', () => {
    const baseElement = renderProvider(
      <ExampleScreenTest fetchData={submitSpy} />
    );
    expect(baseElement).toMatchSnapshot();
  });

  it('should fetch the user data on mount', () => {
    renderProvider(<ExampleScreenTest fetchData={submitSpy} />);
    expect(submitSpy).toHaveBeenCalled();
  });

  it('should render container', () => {
    const { getByTestId } = renderProvider(
      <ExampleScreenTest fetchData={submitSpy} />
    );
    expect(getByTestId('container')).toBeTruthy();
    expect(submitSpy).toHaveBeenCalled();
  });

  it('should wait for data', () => {
    const { getByTestId } = renderProvider(
      <ExampleScreenTest fetchData={submitSpy} data={{ items: undefined }} />
    );
    expect(getByTestId('wait').type).toBe('Text');
    expect(submitSpy).toHaveBeenCalled();
  });

  it('should render repo if data is present', () => {
    const { getByTestId } = renderProvider(
      <ExampleScreenTest fetchData={() => 0} data />
    );
    expect(getByTestId('example-container-content').type).toBe('View');
    expect(submitSpy).toHaveBeenCalled();
  });
});
