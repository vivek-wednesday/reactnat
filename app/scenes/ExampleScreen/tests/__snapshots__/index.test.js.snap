// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExampleScreen /> Container tests should render and match the snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "marginBottom": 30,
        "marginLeft": 30,
        "marginRight": 30,
        "marginTop": 30,
      },
    ]
  }
  testID="container"
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="small"
    testId="loader"
  />
</View>
`;
